crates/stc_ts_file_analyzer/src/analyzer/types/mod.rs

```diff
@@ -2109,21 +2130,94 @@ impl Analyzer<'_, '_> {
+                Ok(Type::StringMapping(StringMapping {
                     span,
                     kind: ty.kind.clone(),
                     type_args: TypeParamInstantiation {
                         span: ty.type_args.span,
                         params: vec![arg],
                     },
                     metadata: ty.metadata,
-                }));
+                }))
             }
 
-            _ => {}
+            _ => Ok(Type::StringMapping(ty.clone())),
+        };
+
+        #[allow(clippy::question_mark)]
+        if let Ok(ref ty) = normalized_ty {
+            if let Type::StringMapping(ty) = ty.normalize() {
+                match ty.type_args.params[0].normalize() {
+                    Type::Ref(ref_ty) => {
+                        if let RTsEntityName::Ident(r_ident) = &ref_ty.type_name {
+                            if let Ok(Some(ty_found)) = self.find_type(&r_ident.into()) {
+                                let ty_found = &ty_found.into_iter().map(|v| v.into_owned()).collect::<Vec<Type>>()[0];
+
+                                if let Type::Alias(alias) = ty_found.normalize() {
+                                    return self.expand_intrinsic_types(
+                                        ty.span(),
+                                        &StringMapping {
+                                            span: ty.span(),
+                                            kind: ty.clone().kind,
+                                            type_args: TypeParamInstantiation {
+                                                span: alias.span,
+                                                params: vec![*alias.ty.clone()],
+                                            },
+                                            metadata: ty.metadata,
+                                        },
+                                        span_for_validation,
+                                    );
+                                } else {
+                                    return self.expand_intrinsic_types(
+                                        ty.span(),
+                                        &StringMapping {
+                                            span: ty.span(),
+                                            kind: ty.clone().kind,
+                                            type_args: TypeParamInstantiation {
+                                                span: ty_found.span(),
+                                                params: vec![ty_found.clone()],
+                                            },
+                                            metadata: ty.metadata,
+                                        },
+                                        span_for_validation,
+                                    );
+                                }
+                            }
+                        }
+                    }
+                    Type::Alias(alias) => {
```

Comment:
This is wrong

Reply:
This should be unreachable

Reply:
This variant should be processed by `expand()`, `expand_top_ref` or `normalize()`